enderio-machines/src/main/java/com/enderio/machines/common/blockentity/solar/SolarPanelBlockEntity.java

```diff
@@ -115,18 +114,12 @@ public int getGenerationRate() {
             return 0;
         }
 
-        boolean day;
-        boolean night;
-        if (soulData != null) {
-            day = soulData.daytime();
-            night = soulData.nighttime();
-        } else {
-            day = level.isDay();
-            night = level.isNight();
-        }
+        boolean day = level.isDay();
+        boolean night = level.isNight();
+        boolean soulHasNight = soulData != null && soulData.nighttime();
 
         float outputScale = 0;
-        if (day && (night || hasLiquidSunshine())) {
+        if (soulHasNight || hasLiquidSunshine()) {
```

Comment:
Looks like we should be checking if night == true as well as the soul doing nighttime, as currently i think this'll make it work all the time

Reply:
Yeah, I wasn't sure about this either. It seemed odd that it would work better at night than in the day so it was intentional.
I think it's probably better to change it so night and day work the same (with ramping output etc).
You can ignore this fix completely until I am able to do that.
I should make sure I understand the intent first before 'fixing' it.